### PR TITLE
feat(function): Add Spark dayname function

### PR DESCRIPTION
**Pull request by yaooqinn**
Implements the `dayname()` function for Spark SQL compatibility that returns the three-letter abbreviated day name (Sun, Mon, Tue, Wed, Thu, Fri, Sat) from a given date.

### CODE REVIEW

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp [hunk 1]
```diff
@@ -658,6 +658,24 @@ TEST_F(DateTimeFunctionsTest, weekdayDate) {
   EXPECT_EQ(4, weekday(parseDate("1582-10-15")));
 }
 
+TEST_F(DateTimeFunctionsTest, dayNameDate) {
+  const auto dayName = [&](std::optional<int32_t> date) {
+    return evaluateOnce<std::string>("dayname(c0)", DATE(), date);
+  };
+
+  EXPECT_EQ(std::nullopt, dayName(std::nullopt));
```
Comment (zhli1142015) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp:666]: nit: I think there is no need to test nulls, since the implementation of this function follows the default null behavior.

Comment (yaooqinn) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp:666]:
```suggestion
```

Comment (yaooqinn) [reply velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp:666]: Make sense. Thank you @zhli1142015

File: velox/docs/spark_functions.rst [hunk 2]
```diff
@@ -130,9 +130,10 @@ for :doc:`all <functions/spark/coverage>` functions.
     :spark:func:`date_trunc`                     :spark:func:`map_filter`                     :spark:func:`upper`
     :spark:func:`datediff`                       :spark:func:`map_from_arrays`                :spark:func:`url_decode`
     :spark:func:`day`                            :spark:func:`map_keys`                       :spark:func:`url_encode`
-    :spark:func:`dayofmonth`                     :spark:func:`map_values`                     :spark:func:`uuid`
-    :spark:func:`dayofweek`                      :spark:func:`map_zip_with`                   :spark:func:`varchar_type_write_side_check`
-    :spark:func:`dayofyear`                      :spark:func:`mask`                           :spark:func:`week_of_year`
+    :spark:func:`dayname`                        :spark:func:`map_values`                     :spark:func:`uuid`
```
Comment (jinchengchenghh) [inline velox/docs/spark_functions.rst]: Please don't update this file, weekly job will does the update

Comment (yaooqinn) [reply velox/docs/spark_functions.rst]: This helps a lot. Thank you @jinchengchenghh

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp [hunk 3]
```diff
@@ -658,6 +658,24 @@ TEST_F(DateTimeFunctionsTest, weekdayDate) {
...
+  EXPECT_EQ("Thu", dayName(0)); // 1970-01-01
+  EXPECT_EQ("Sun", dayName(parseDate("2023-08-20")));
+  EXPECT_EQ("Mon", dayName(parseDate("2023-08-21")));
+  EXPECT_EQ("Tue", dayName(parseDate("2023-08-22")));
+  EXPECT_EQ("Wed", dayName(parseDate("2023-08-23")));
+  EXPECT_EQ("Thu", dayName(parseDate("2023-08-24")));
+  EXPECT_EQ("Fri", dayName(parseDate("2023-08-25")));
+  EXPECT_EQ("Sat", dayName(parseDate("2023-08-26")));
+  EXPECT_EQ(
+      "Fri", dayName(parseDate("1582-10-15"))); // Gregorian calendar start
```
Comment (jinchengchenghh) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp]: The comment should end with `.`

Comment (yaooqinn) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp]:
```suggestion
      "Fri", dayName(parseDate("1582-10-15"))); // Gregorian calendar start.
```
